src/containers/Tenant/QueryEditor/SavedQueries/SavedQueries.tsx

```diff
@@ -0,0 +1,151 @@
+    onConfirmClick: VoidFunction;
+}
+
+const DeleteDialog = ({visible, queryName, onCancelClick, onConfirmClick}: DeleteDialogProps) => {
+    return (
+        <Dialog
+            open={visible}
+            hasCloseButton={false}
+            size="s"
+            onClose={onCancelClick}
+            onEnterKeyDown={onConfirmClick}
+        >
+            <Dialog.Header caption={i18n('delete-dialog.header')} />
+            <Dialog.Body className={b('dialog-body')}>
+                {i18n('delete-dialog.question')}
+                <span className={b('dialog-query-name')}>{` ${queryName}?`}</span>
+            </Dialog.Body>
+            <Dialog.Footer
+                textButtonApply={i18n('delete-dialog.delete')}
+                textButtonCancel={i18n('delete-dialog.cancel')}
+                onClickButtonCancel={onCancelClick}
+                onClickButtonApply={onConfirmClick}
+            />
+        </Dialog>
+    );
+};
+
+interface SavedQueriesProps {
+    savedQueries: SavedQuery[];
+    changeUserInput: (value: {input: string}) => void;
+    onDeleteQuery: (queryName: string) => void;
+}
+
+export const SavedQueries = ({savedQueries, changeUserInput, onDeleteQuery}: SavedQueriesProps) => {
+    const dispatch = useDispatch();
+
+    const [isDeleteDialogVisible, setIsDeleteDialogVisible] = useState(false);
+    const [queryNameToDelete, setQueryNameToDelete] = useState<string>('');
+
+    const closeDeleteDialog = () => {
+        setIsDeleteDialogVisible(false);
+        setQueryNameToDelete('');
+    };
+
+    const onCancelDeleteClick = () => {
+        closeDeleteDialog();
+    };
+
+    const onConfirmDeleteClick = () => {
+        closeDeleteDialog();
+        onDeleteQuery(queryNameToDelete);
+        setQueryNameToDelete('');
+    };
+
+    const onQueryClick = (queryText: string, queryName: string) => {
+        changeUserInput({input: queryText});
+        dispatch(setQueryNameToEdit(queryName));
+        dispatch(setQueryTab(TENANT_QUERY_TABS_ID.newQuery));
+    };
+
+    const onDeleteQueryClick = (queryName: string) => {
+        return (event: MouseEvent) => {
+            event.stopPropagation();
+            setIsDeleteDialogVisible(true);
+            setQueryNameToDelete(queryName);
+        };
+    };
+
+    const columns: Column<SavedQuery>[] = [
+        {
+            name: 'name',
+            header: 'Name',
+            render: ({row: query}) => <div className={b('query-name')}>{query.name}</div>,
+            width: 200,
+        },
+        {
+            name: 'body',
+            header: 'Query Text',
+            render: ({row: query}) => (
+                <div className={b('query')}>
+                    <div className={b('query-body')}>
+                        <TruncatedQuery value={query.body} maxQueryHeight={MAX_QUERY_HEIGHT} />
+                    </div>
+                    <span className={b('control-delete')} onClick={onDeleteQueryClick(query.name)}>
+                        <IconWrapper
```

Comment:
Icon looks pale and poorly visible. May be change it's color to primary and set size 16?

Reply:
As user I expect `edit` button as well. Because it's not clear that I will be able to edit my query after click the row. What do you think about adding it? And may be default action should be `edit`, not `save as new`

Reply:
Missed comment about buttons colors, I'll update them as well

Reply:
Updated buttons colors as well

Reply:
Where should be such a button? And do we really need it if we can use default button behaviour via option in dropdown menu?

Reply:
Yeah it seems it could be enough!